### PR TITLE
travis: add support groonga-latest

### PR DESCRIPTION
**Pull request by komainu8**

### CODE REVIEW

**Comment by komainu8** [comment]
@kenhys Could you review this PR, please?

**Comment by kenhys** [comment]
I'll look it.

File: .travis.yml [hunk 1]
```diff
@@ -6,15 +6,25 @@ rvm:
...
 #   allow_failures:
 #     - rvm: ruby-head
 before_install:
-  - curl --silent --location https://github.com/groonga/groonga/raw/master/data/travis/setup.sh | sh
   - |
-    if [ "${GROONGA_MASTER}" != "yes" ]; then
-      sudo apt-get -qq -y install groonga-tokenizer-mecab
+    if [ "${GROONGA_LATEST}" != "yes" ]; then
+      curl --silent --location https://github.com/groonga/groonga/raw/master/data/travis/setup.sh | sh
+    - |
```
Comment (kenhys) [inline .travis.yml]: Is this `- | `  needed? (maybe I'm not familiar with it, so just ignore it if it is expected)

**Comment by komainu8** [comment]
@kenhys Thank you for your comment.
I've fixed it.

**Comment by komainu8** [comment]
@kou Could you review this PR, please.

File: .travis.yml [hunk 2]
```diff
@@ -6,15 +6,22 @@ rvm:
...
 #     - rvm: ruby-head
 before_install:
-  - curl --silent --location https://github.com/groonga/groonga/raw/master/data/travis/setup.sh | sh
   - |
-    if [ "${GROONGA_MASTER}" != "yes" ]; then
-      sudo apt-get -qq -y install groonga-tokenizer-mecab
+    if [ "${GROONGA_LATEST}" != "yes" ]; then
+      curl --silent --location https://github.com/groonga/groonga/raw/master/data/travis/setup.sh | sh
+      if [ "${GROONGA_MASTER}" != "yes" ]; then
+        sudo apt-get -qq -y install groonga-tokenizer-mecab;
```
Comment (kou) [inline .travis.yml]: The last `;` is needed?

File: .travis.yml [hunk 3]
```diff
@@ -6,15 +6,22 @@ rvm:
...
+      fi
     fi
   - gem update bundler
+after_failure:
+  - cat ./ext/groonga/mkmf.log
```
Comment (kou) [inline .travis.yml]: `./` is needless.

File: .travis.yml [hunk 4]
```diff
@@ -6,15 +6,22 @@ rvm:
   - 2.4
   - 2.5
   - 2.6
-# env:
-#   - GROONGA_MASTER=yes
+env:
+  matrix:
+    - GROONGA_MASTER=no GROONGA_LATEST=yes
+    - GROONGA_MASTER=yes GROONGA_LATEST=no
+    - GROONGA_MASTER=no GROONGA_LATEST=no
```
Comment (kou) [inline .travis.yml]: Should we run these patterns for all Rubies? The auto built Groonga isn't depended on Ruby version.

```diff
diff --git a/.travis.yml b/.travis.yml
index 4b19b2ff..84cc25f0 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -2,19 +2,25 @@ sudo: required
 notifications:
   webhooks:
     - https://webhook.commit-email.info/
-rvm:
-  - 2.4
-  - 2.5
-  - 2.6
 # env:
 #   - GROONGA_MASTER=yes
-# matrix:
-#   allow_failures:
-#     - rvm: ruby-head
+matrix:
+  - name: "Ruby 2.4"
+    rvm: 2.4
+  - name: "Ruby 2.5"
+    rvm: 2.5
+  - name: "Ruby 2.6"
+    rvm: 2.6
+  - name: "Auto Groonga build"
+    rvm: 2.6
+    env:
+      - INSTALL_GROONGA=no
 before_install:
-  - curl --silent --location https://github.com/groonga/groonga/raw/master/data/travis/setup.sh | sh
   - |
-    if [ "${GROONGA_MASTER}" != "yes" ]; then
-      sudo apt-get -qq -y install groonga-tokenizer-mecab
+    if [ "${INSTALL_GROONGA}" != "no" ]; then
+      curl --silent --location https://github.com/groonga/groonga/raw/master/data/travis/setup.sh | sh
+      if [ "${GROONGA_MASTER}" != "yes" ]; then
+        sudo apt-get -qq -y install groonga-tokenizer-mecab
+      fi
     fi
   - gem update bundler
```

**Comment by komainu8** [comment]
@kou Thank you for your comments.
I've fixed them.

**Pull request merged**